image_writers_guide/guidelines.adoc

```diff
@@ -5,143 +5,186 @@
 
+If you later release an incompatible update, then you should switch to a new tag, for example _foo:v2_. This allows downstream consumers to move up to the new version at will, but not be inadvertently broken by the new incompatible image. Of course, any downstream consumer who chooses to use _foo:latest_ will be taking on the risk of any incompatible changes introduced.
 
-=== Use exec in wrapper scripts
-See http://www.projectatomic.io/docs/docker-image-author-guidance/, subsection _Always `exec` in Wrapper Scripts_
+*Avoiding Multiple Processes*
 
-Also be aware that your process runs as PID 1 when running in a Docker container.  This means that if your main process terminates, the entire container will be stopped, killing any child processes you may have launched from your PID 1 process.  Additional implications to be aware of are documented http://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/[here].  For a deep dive on PID 1 and init systems, you can read https://felipec.wordpress.com/2013/11/04/init/[this] as well.
+Do not start multiple services, such a database and [sysitem]#sshd#, inside one container. Containers are lightweight and can be easily linked together for multiprocess orchestration. OpenShift allows you to easily collocate and comanage related images by grouping them into a single pod.
 
+This collocation ensures they share a network namespace and storage for communication. It also makes updates less disruptive as each image can be updated less frequently and independently. Having a single process also makes signal handling flows clearer as you do not need to manage routing signals to spawned processes.
 
-=== Clean temporary files
-Every temporary files you create during the build process should be removed. This includes any files added with the ADD command too.  For example it is strongly recommended you run yum clean after performing yum install operations.  If you craft your `RUN` statement as follows, you an prevent the yum cache from ending up in an image layer:
+*Using `exec` in Wrapper Scripts*
 
-	RUN yum -y install mypackage && yum -y install myotherpackage && yum clean all -y
+See the "Always `exec` in Wrapper Scripts" section of the http://www.projectatomic.io/docs/docker-image-author-guidance[Project Atomic documentation] for more information.
+
+Also be aware that your process runs as PID 1 when running in a Docker container. This means that if your main process terminates, the entire container is stopped, killing any child processes you may have launched from your PID 1 process.
+
+See the http://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/["Docker and the PID 1 zombie reaping problem"] blog article for additional implications. See the https://felipec.wordpress.com/2013/11/04/init/["Demystifying the init system (PID 1)"] blog article for a deep dive on PID 1 and [sysitem]#init# systems, as well.
+
```

Comment:
[Suggestion]
See the https://felipec.wordpress.com/2013/11/04/init/["Demystifying the init system (PID 1)"] blog article for a deep dive on PID 1 and [sysitem]#init# systems, as well. -->> Also see the https://felipec.wordpress.com/2013/11/04/init/["Demystifying the init system (PID 1)"] blog article for a deep dive on PID 1 and [sysitem]#init# systems.